stacks-signer/src/client/stackerdb.rs

```diff
@@ -79,23 +121,50 @@ impl StackerDB {
         message: SignerMessage,
     ) -> Result<StackerDBChunkAckData, ClientError> {
         let message_bytes = message.serialize_to_vec();
-        let slot_id = message.slot_id(self.signer_id);
-
+        let msg_id = message.msg_id();
+        let signer_set = self.signer_set;
+        let slot_id = self.signer_slot_id;
         loop {
-            let slot_version = *self.slot_versions.entry(slot_id).or_insert(0) + 1;
+            let slot_version =
```

Comment:
Could you not do 

```rust
let slot_version = {
    let versions = self.slot_versions.entry((signer_set, msg_id))
        .or_insert_with(|| HashMap::new());
    *versions.entry(slot_id).or_insert(0).saturating_add(1)
};
```
And remove the following NOTE this will always execute per the above?